docs/funnel.md

```diff
@@ -0,0 +1,228 @@
+# funnels [<sup><mark>experimental</mark></sup>](/docs/README.md#tag-notes)
+
+- Current status:	prototype
+- Effort remaining:	lots
+
+## Purpose
+
+TODO
+
+## Use cases
+
+TODO
+
+### Test Data
+
+Examples below are tested against the following data:
+
+```SQL ,non-transactional
+DROP TABLE IF EXISTS funnel_test;
+CREATE TABLE funnel_test(
+	"time" TIMESTAMPTZ,
+	"user" INTEGER,
+	"event" TEXT);
+INSERT INTO funnel_test VALUES
+	('2020-01-01 00:00:00+00', 1, 'LOGIN'),
+	('2020-01-02 00:00:00+00', 1, 'LOGIN'),
+	('2020-01-02 01:00:00+00', 1, 'SEARCH'),
+	('2020-01-02 03:00:00+00', 1, 'BUY'),
+
+	('2020-01-03 00:00:00+00', 2, 'LOGIN'),
+	('2020-01-03 01:00:00+00', 2, 'SEARCH'),
+	('2020-01-03 01:30:00+00', 2, 'BUY'),
+	('2020-01-20 00:00:00+00', 2, 'LOGIN'),
+
+	('2020-02-01 01:00:00+00', 3, 'SEARCH'),
+	('2020-02-14 01:30:00+00', 3, 'BUY'),
+
+	('2020-02-15 00:00:00+00', 4, 'LOGIN'),
+	('2020-02-28 00:00:00+00', 4, 'LOGIN');
+```
+
+### within_interval
+
+Name is terrible, this is just a first test.
+
+Looking for rows where one event occurs within an interval of another, e.g.
+- two LOGIN events within one week
+- BUY event within one hour of SEARCH
+
+#### two LOGIN events within one week
+
+```SQL
+SELECT toolkit_experimental.within_interval(
+	'LOGIN', 'LOGIN', '1 week',
+	toolkit_experimental.funnel_agg("user", event, time))
```

Comment:
I don't know why user is included in the aggregate, I think it should probably just be in the group by so: 

`SELECT user, funnel_agg(event, time) FROM funnel_test GROUP BY user`

Reply:
I also wonder whether we want to add an option to the aggregate to deduplicate the underlying data or not...or maybe we could just compress the whole thing then it wouldn't matter so much....